### PR TITLE
Rule constraints referencing corrections

### PR DESCRIPTION
**Pull request by AlexBrad1eyCT**
fixed the references for sub templates in the IfcDoc representation, and othe minor corrections.

### CODE REVIEW

**Comment by AlexBrad1eyCT** [comment]
@jmirtsch or @SergejMuhic can you review and approve this ASAP so we have a douvle check for errors and mistakes

**Review by jmirtsch — APPROVED** [review]
It looks ok to me, but not an easy one to manually review.  Hopefully @SergejMuhic can also review.

**Comment by AlexBrad1eyCT** [comment]
@jmirtsch yeah bit of a difficult one, it all loads correctly in IfcDoc that is the main thing!

File: IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/Documentation.md [hunk 1]
```diff
@@ -1,4 +1,3 @@
-
 The _Spatial Interference with Zone_ concept is an extension of the _Spatial Inteference_ concept which defines the relationship of spatial elements, such as facilities (_IfcFacility_ & specialised subtypes) or faclity parts (_IfcFacilityPart_ with domain specific predefined types) that interfere or interface with other spatial elements across discipline spatial hierarchy branches, and extends the dataset with an related inteference _IfcSpatialZone_ via the _IfcRelInterferesElements.InterferenceSpace_ attribute. The realizing _IfcSpatialZone_ should always have the _PredefinedType_ value set to INTERFERENCE.
```
Comment (pjanck) [inline IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/Documentation.md:1]:
```suggestion
The _Spatial Interference with Zone_ concept is an extension of the _Spatial Interference_ concept which defines the relationship of spatial elements, such as facilities (_IfcFacility_ & specialised subtypes) or facility parts (_IfcFacilityPart_ with domain specific predefined types) that interfere or interface with other spatial elements across discipline spatial hierarchy branches, and extends the dataset with a related interference _IfcSpatialZone_ via the _IfcRelInterferesElements.InterferenceSpace_ attribute. The realizing _IfcSpatialZone_ should always have the _PredefinedType_ value set to INTERFERENCE.
```

Also, I'm unsure about the coding of the `&` in xml code.

File: IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/Documentation.md [hunk 2]
```diff
@@ -1,4 +1,3 @@
...
 
 The addition of a realizing _ifcSpatialZone_ extends the functionality of this relationship with the following:
```
Comment (pjanck) [inline IFC4x3/Templates/Object Connectivity/Spatial Structure/Spatial Interference/Spatial Interference With Zones/Documentation.md:3]:
```suggestion
The addition of a realizing _IfcSpatialZone_ extends the functionality of this relationship with the following:
```

**Review by pjanck — CHANGES_REQUESTED** [review]
Lots of deletions and indentation changes. 

I just saw some typos and improved upon them here - did not see it necessary to open another pull request for that.

**Pull request merged**